fhevm/tee_interpreter.go

```diff
@@ -405,57 +405,33 @@ func marshalTfheType(value any, typ tfhe.FheUintType) ([]byte, error) {
 func unmarshalTfheType(data []byte, typ tfhe.FheUintType) (uint64, error) {
 	switch typ {
 	case tfhe.FheBool:
-		if len(data) < 1 {
-			return 0, fmt.Errorf("data too short for type %s", typ)
-		}
-		if data[0] == 0 {
+		if data[len(data)-1] == 0 {
 			return 0, nil
 		} else {
 			return 1, nil
 		}
-	case tfhe.FheUint4:
-		if len(data) < 1 {
-			return 0, fmt.Errorf("data too short for type %s", typ)
-		}
-		return uint64(data[0] % 16), nil
-	case tfhe.FheUint8:
-		if len(data) < 1 {
-			return 0, fmt.Errorf("data too short for type %s", typ)
-		}
-		return uint64(data[0]), nil
+	case tfhe.FheUint4, tfhe.FheUint8:
```

Comment:
Also not related to this line, but in tee_cast.go, remove these 3 lines:

```go
	if ciphertext.FheUintType == castToType {
		return nil, errors.New("casting to same type is not supported")
	}
```

Reply:
For fhe operations, casting to same type is not allowed. https://github.com/Inco-fhevm/fhevm-go/blob/main/fhevm/tfhe/tfhe_ciphertext.go#L2024
So this is just for compatibility.

Reply:
Ah, right, I missed this. Okay let's keep it.